### DISCUSSION TITLE
How to switch the chip on which the program runs？

### DISCUSSION THREAD

**Discussion by tibaiwan2333** [grblHAL/ESP32#131] [Q&A]
I'm looking to run this program on an ESP32-S3. Could you please advise on what changes I need to make to the project to ensure it works correctly on this chip? Any guidance or suggestions would be greatly appreciated.

**Comment by terjeio** [comment]
Delete (or rename) _sdkconfig_ and rename _sdkconfig.s3_ to _sdkconfig_, they are located in the project root directory.
No other changes than creating or selecting an existing board map for the S3 is needed. Board maps made for ESP32 may or may not work, likely pin assignments has to be changed.

I am using Espressif-IDE for development, there I change the target to _esp32s3_:

![image](https://github.com/user-attachments/assets/33be176e-0e36-4db8-9545-06edc7a4e490)

I have not compiled for the ESP32-S3 using other IDEs or ESP-IDF yet so I do not know how to change the target MCU for them.

**Reply by tibaiwan2333** [reply]
Thank you very much for your assistance, Sir. I have installed IDF 4.3 as suggested by the project documentation; however, I noticed that this installation does not include an IDE. Consequently, I visited the ESP32 official website to install IDF along with an IDE. There, I found that only version 5.1 or higher is available. When I opened my project in the provided IDE and attempted to compile it, I encountered numerous errors. As a result, I am still unable to find a way to flash the firmware onto the ESP32-S3.Could you kindly guide me on how to proceed? I would greatly appreciate any advice or suggestions you could offer to help resolve this issue.

**Reply by terjeio** [reply]
> Could you kindly guide me on how to proceed?

Ask Espressif from where to download v4.3.
I am not yet prepared to fix/reverify against v5 since they are introducing breaking changes with every major release that can be very hard to debug. And debugging ESP32 code is very time consuming.

**Reply by luc-github** [reply]
I do not know if it can help but on my project using idf I use this command on my ci to switch from esp32 to esp32s3:
https://github.com/luc-github/ESP3D-TFT/blob/main/.github/ci/ESP32S3_FREENOVE.sh
```
idf.py fullclean
idf.py -DIDF_TARGET=esp32s3 reconfigure
idf.py build
```

**Reply by gwcSoftware** [reply]
I know this is a little late and hope you found a resolution. The following is a link to the Espressif downloads for previous versions.

https://[dl.espressif.com/dl/esp-idf/?idf=4.4](https://dl.espressif.com/dl/esp-idf/?idf=4.4)

I have been using version 4.4 to build ESP32.

**Reply by terjeio** [reply]
I'll be moving the Web Builder to a new server next week, will try to add S3 support then.

**Reply by terjeio** [reply]
S3 support added for two board maps in the Web Builder - MKS DLC32 MAX and a generic board.
I have only bench tested these with no motors connected so please report any issues.
Note that both boards defaults to comms via the native USB port. Select _(Virtual) UART_ in the _Connection_ dropdown if the USB <>UART chip is to be used.
